sockeye/lr_scheduler.py

```diff
@@ -212,21 +254,17 @@ def get_lr_scheduler(scheduler_type: str,
 
     :raises: ValueError if unknown scheduler_type
 
-    :return: Learning rate scheduler.
+    :return: Tuple of LearningRateScheduler class and kwargs dictionary.
     """
     if scheduler_type is None or scheduler_type == C.LR_SCHEDULER_NONE:
-        return None
+        return None, {}
     if scheduler_type == C.LR_SCHEDULER_INV_SQRT_DECAY:
-        return LearningRateSchedulerInvSqrtDecay(base_lr=base_learning_rate, warmup=learning_rate_warmup,
-                                                 t_scale=learning_rate_t_scale)
+        return LearningRateSchedulerInvSqrtDecay, {'base_lr': base_learning_rate, 'warmup': learning_rate_warmup}
     if scheduler_type == C.LR_SCHEDULER_LINEAR_DECAY:
-        check_condition(max_updates is not None,
-                        "The total number of training updates (--max-updates) must be specified when using the linear "
-                        "decay learning rate scheduler.")
-        return LearningRateSchedulerLinearDecay(base_lr=base_learning_rate,
-                                                total_steps=max_updates,
-                                                warmup=learning_rate_warmup,
-                                                t_scale=learning_rate_t_scale)
+        assert max_updates is not None, "The total number of training updates (--max-updates) must be specified when " \
```

Comment:
why not keep using check_condition (which returns a nicer looking error than `assert`)?

Reply:
Fixed